step.sh

```diff
@@ -1,27 +1,40 @@
 #!/bin/bash
 set -e
 
-echo "Checking status of collector at $collector_url..."
-result=$(curl --silent "$collector_url/micro/all")
+echo "Checking status of collector at ${CP_BITRISE_SNOWPLOW_MICRO_COLLECTOR_URL}/micro/all"
+result=$(curl --silent "${CP_BITRISE_SNOWPLOW_MICRO_COLLECTOR_URL}/micro/all")
+
+echo result
 
 # Parse the results
-total=$(jq '.total' <<< "$result")
-good=$(jq '.good' <<< "$result")
-bad=$(jq '.bad' <<< "$result")
+total=$(jq -r '.total' <<< "$result")
+good=$(jq -r '.good' <<< "$result")
+bad=$(jq -r '.bad' <<< "$result")
 
 # Export them to variables
 echo "Parsed results (total: $total, good: $good, bad: $bad)"
 envman add --key SNOWPLOW_MICRO_COLLECTOR_RESULTS_TOTAL --value "$total"
 envman add --key SNOWPLOW_MICRO_COLLECTOR_RESULTS_GOOD --value "$good"
 envman add --key SNOWPLOW_MICRO_COLLECTOR_RESULTS_BAD --value "$bad"
 
-# TODO: Do a lot more.. Export the bad results (log files? test results?)
-# TODO: Offer an option to reset?
-# TODO: Maybe use a better language, Go? or Ruby?
+# TODO: In the future
+# We need to do bad events parsing according to https://docs.snowplowanalytics.com/docs/managing-data-quality/testing-and-qa-workflows/set-up-automated-testing-with-snowplow-micro/#micro-bad
+# Then create a junit report file that lists each issue as a failure, write it to $BITRISE_TEST_RESULT_DIRCreate a junit report file that lists each issue as a failure, write it to $BITRISE_TEST_RESULT_DIR
+# It will be sharable script codes among other clients like `Android` and `Web`
 
 # Fail if bad is != 0
-if [ "$bad" != "0" ];
+if [[ $total == "0" ]]
+then
+  echo "There are no events posted to Snowplow Micro, text failed."
+  exit 1
+elif [[ $bad == "0" ]]
 then
+  echo "Snowplow test was successfully. :tada"
+else
   echo "Failing step because Snowplow reports more than zero bad events ($bad)"
+  
+  # Export the bad json into a file
+  curl --silent "${CP_BITRISE_SNOWPLOW_MICRO_COLLECTOR_URL}/micro/bad" | json_pp >> $BITRISE_DEPLOY_DIR/snowplow_bad.json
```

Comment:
TIL `json_pp`!

Is this also available on linux for when we use this in the Android workflow too? If not, we can also use `jq` for consistency 

```suggestion
  curl --silent "${CP_BITRISE_SNOWPLOW_MICRO_COLLECTOR_URL}/micro/bad" | jq '.' >> $BITRISE_DEPLOY_DIR/snowplow_bad.json
```

Reply:
We have tested on both Linux and Mac OS on bitrise, json_pp can be used both. Although, I got to know, `jq '.'` do the same trick now 😂 TIL

Reply:
If we use `json_pp`, I _think_ we are also supposed to list it as a dependency in **step.yml** 🤔 But I'm not 100% sure

Also we need to use `$collector_url` here 😬 

```suggestion
  curl --silent "$collector_url/micro/bad" | json_pp >> $BITRISE_DEPLOY_DIR/snowplow_bad.json
```

Reply:
> If we use json_pp, I think we are also supposed to list it as a dependency in step.yml 🤔 But I'm not 100% sure

I think you are right here, although when we test for now json_qq is working, we cannot guarantee it will work in future OS updates

Reply:
I have updated to use `jq '.' ` instead of json_pp 😄 Thank you Liam